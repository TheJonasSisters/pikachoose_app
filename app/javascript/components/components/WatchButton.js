import React from "react"
import { Button } from "reactstrap"

const WatchButton = (props) => {
  const { filmDecision, renderDecisionBox } = props

  const addHistory = (event) => {
    event.preventDefault()
    renderDecisionBox()

    let newHistory = {
      film_id: filmDecision.id,
      title: filmDecision.title,
      overview: filmDecision.overview,
      vote_average: filmDecision.vote_average,
      release_date: filmDecision.release_date,
      comment: "Add comment"
    }

    console.log("newHistory:",newHistory)

    fetch("/histories",
    {
      method: 'POST',
      body: JSON.stringify(newHistory),
      headers: {
        'Content-Type': 'application/json'
      }
    }).then((response) => {
      if (response.ok) {
        return response.json()
      }
    })
  }

  return (
    <div id="watch-btn">
      <Button type="button" size="lg" onClick={addHistory}><h5>What to Watch</h5></Button>
    </div>
  )
}

export default WatchButton
