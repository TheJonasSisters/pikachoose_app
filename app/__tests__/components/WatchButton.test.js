import React from "react";
import { Button } from "reactstrap"
import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Route, Link, Switch } from "react-router-dom";
import Enzyme, {shallow, mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import WatchButton from "../../javascript/components/components/WatchButton";

Enzyme.configure({ adapter: new Adapter() });

describe('WatchButton component', () => {
  it('Test click event', () => {
    const mockCallBack = jest.fn();
    const button = mount((<Button onClick={mockCallBack}>Ok!</Button>));
    button.find('Button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});