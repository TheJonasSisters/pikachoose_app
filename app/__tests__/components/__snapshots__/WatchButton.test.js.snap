// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchButton component Component: WatchButton WatchButton renders with default props 1`] = `ShallowWrapper {}`;
